klog.go

```diff
@@ -748,6 +748,49 @@ func (l *loggingT) printWithFileLine(s severity, logr logr.InfoLogger, file stri
 	l.output(s, logr, buf, file, line, alsoToStderr)
 }
 
+// printS if loggr is specified, no need to output with logging module. If
+// err arguments is specified, will call logr.Error, or output to errorLog severity
+func (l *loggingT) printS(err error, loggr logr.Logger, msg string, keysAndValues ...interface{}) {
+	if loggr != nil {
+		if err != nil {
+			loggr.Error(err, msg, keysAndValues)
+		} else {
+			loggr.Info(msg, keysAndValues)
+		}
+		return
+	}
+	b := &bytes.Buffer{}
+	b.WriteString(fmt.Sprintf("%q", msg))
+	if err != nil {
+		b.WriteByte(' ')
+		b.WriteString(fmt.Sprintf("err=%q", err.Error()))
+	}
+	kvListFormat(b, keysAndValues...)
+	var s severity
+	if err == nil {
+		s = infoLog
+	} else {
+		s = errorLog
+	}
+	l.printDepth(s, logging.logr, 1, b)
+}
+
+const missingValue = "(MISSING)"
+
+func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
```

Comment:
I appreciate the test cases, but I was looking for a unit test of this function, passing in all varieties of input, checking for known results

Reply:
responded